### ISSUE TITLE
publishing with scripts - problem with reply

### ISSUE THREAD

**Issue by gatovelho** [kamilfb/mqtt-spy#72]
Hello,

I am using a simplereply2.js script for auto feedback/callback on a message topic.
It is being declared in the connection manager, in the subscriptions/topic/script  entry tab, by 
complete name (including .js)

This script, apparently, is not called/executed when it resides in the configuration/scripts directory , 
to where the scripts are saved by default, but is properly executed when located in the current 
mqtt-spy directory.  

Besides being used as a callback, it publishes log in a  /log topic I have defined.  
All goes well from the application directory, but not from the publishing script directory.

Best regards,
Carlos

**Comment by kamilfb** [comment]
@gatovelho, are you getting any exceptions logged in the mqtt-spy log file?

**Comment by gatovelho** [comment]
Clean log.

[mqtt-spy.zip](https://github.com/kamilfb/mqtt-spy/files/202293/mqtt-spy.zip)

For this log, the reply script was put in the publish directory (the fail case)

If you want I can also send log for the other setup (simplerepl2 on the app dir)

**Comment by kamilfb** [comment]
Thank you! I will take a look...

**Comment by gatovelho** [comment]
When the script is on the app directory:
    
    2016-04-04 12:59:06,064 INFO  [JavaFX Application Thread     ]
    [ViewManager                   ]  - Loading configuration file from
    C:\Users\Ulisses\mqtt-spy\mqtt-spy-configuration.xml
    2016-04-04 12:59:06,162 INFO  [JavaFX Application Thread     ]
    [MqttConfigurationUtils        ]  - Auto-complete for server URI
    (192.168.1.3:1883 -> tcp://192.168.1.3:1883)
    2016-04-04 12:59:08,701 INFO  [JavaFX Application Thread     ]
    [BaseScriptManager             ]  - Adding script simplereply2.js
    2016-04-04 12:59:08,983 INFO  [JavaFX Application Thread     ]
    [NewSubscriptionController     ]  - Subscribing to
    /mcu/rgbled_status/
    
    I am adding my configuration file just in case.
    Best Regards
    
    
    
    Às 12:50 de 04-04-2016, Kamil
      Baczkowicz escreveu:
    
    
      Thank you! I will take a look...
      —
        You are receiving this because you were mentioned.
        Reply to this email directly or view
          it on GitHub
      
        
          
          
        
        
      
    
    
    -- 

I’m open to any suggestions, I’ve been wrong before.

**Comment by gatovelho** [comment]
[mqtt-spy-configuration.zip](https://github.com/kamilfb/mqtt-spy/files/202350/mqtt-spy-configuration.zip)

**Comment by gatovelho** [comment]
OK found a solution:
    the script name (simplereply2.js) must be entered in full in
    Subscriptions
    
    (copied from configuration file)
    
    before: 
                <Subscription createTab="true" qos="0"
    topic="/mcu/rgbled_status/" scriptFile="simplereply2.js"/>
    after:
                <Subscription createTab="true" qos="0"
    topic="/mcu/rgbled_status/"
    scriptFile="C:\Users\Ulisses\mqtt-spy\simplereply2.js"/>
    
    
    
    
    
    
    Às 12:50 de 04-04-2016, Kamil
      Baczkowicz escreveu:
    
    
      Thank you! I will take a look...
      —
        You are receiving this because you were mentioned.
        Reply to this email directly or view
          it on GitHub
      
        
          
          
        
        
      
    
    
    -- 

I’m open to any suggestions, I’ve been wrong before.

**Comment by gatovelho** [comment]
I just removed some of the content that I  believe was not being useful

**Comment by kamilfb** [comment]
Many thanks for the info - I'll add checking if the scripts exists when setting it up (in the location mqtt-spy thinks it should be in) and also if it is a relative path, looking for it in the configured script directory.

**Comment by gatovelho** [comment]
Okay. And I just removed my config file. Probably not useful for anyone.
Regards!

**Comment by kamilfb** [comment]
Hi - I've added checks to warn the user if the script does not exist. When you type in the name of the script and submit it, it gives you a warning when the path is incorrect. Available in the latest beta.

**Comment by gatovelho** [comment]
```
Thank you for your care.
I will gladly check it. 

Best Regards,

Carlos


Às 20:07 de 13-05-2016, Kamil
  Baczkowicz escreveu:


  Hi - I've added checks to warn the user if the script does not
    exist. When you type in the name of the script and submit it, it
    gives you a warning when the path is incorrect. Available in the
    latest beta.
  —
    You are receiving this because you were mentioned.
    Reply to this email directly or view
      it on GitHub









-- 
```

I’m open to any suggestions, I’ve been wrong before.

**Comment by kamilfb** [comment]
This is now available in 0.5.1. Let me know if you find any further issues!
